posts/views.py

```diff
@@ -37,3 +37,92 @@ def put(self, request):
             serializer.save()
             return Response(serializer.data)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+
+class PostView(views.APIView):
+    #@method_decorator(login_required)
+    def post(self, request):
+        # print(request.data)
+        """
+        Create new categories
+        """
+        categories = request.data.get("categories")
+        if categories is not None:
+            # author has defined categories
+            for cat in categories:
+                # check if category exists
+                try:
+                    cat_obj = Category.objects.get(category=cat)
+                except Category.DoesNotExist:
+                    cat_obj = None
+
+                if cat_obj is None:
+                    Category.objects.create(category=cat)
+
+
+        serializer = PostSerializer(data=request.data, context={'create': False})
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        print(serializer.errors)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request):
+        posts = Post.objects.all()
+        serializer = PostSerializer(posts, many=True)
+        return Response(serializer.data)
+
+
+class PostViewID(views.APIView):
+    def get_post(self, pk):
+        try:
+            return Post.objects.get(pk=pk)
+        except Post.DoesNotExist:
+            raise Http404
+
+    def get(self, request, pk):
+        post = self.get_post(pk)
+        serializer = PostSerializer(post)
+        return Response(serializer.data)
+
+    #@method_decorator(login_required)
+    def delete(self, request, pk):
+        post = self.get_post(pk)
+        post.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class CommentViewList(views.APIView):
+
+    #@method_decorator(login_required)
+    def post(self, request, post_id):
+        serializer = CommentSerializer(data=request.data, context={'create':False, 'post_id':post_id})
+        # print(serializer.initial_data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request, post_id):
```

Comment:
Since we don't have permission for visibility please add a TODO here to verify that before commenting